### DISCUSSION TITLE
V7 needs React 19?

### DISCUSSION THREAD

**Discussion by lucasvsouza28** [next-safe-action/next-safe-action#165] [Q&A] [answered]
I've writed some safe actions with the new v7 release and all works great, however, my Next app cant build anymore.
The build error log says that useActionState is not exported from 'React@18.0.2', wich perfectly true, but the docs says that the lib should workk with React 18.0.2.

**Comment by TheEdoRan** [comment]
Yeah, I'm aware of this error, I'm working on it right now, sorry for the inconvenience.

**Comment by TheEdoRan** [comment, marked as answer]
v7.0.1 should fix this issue, as now `useStateAction` is exported from `next-safe-action/stateful-hooks` path. Please let me know if this works for you as well, thank you.

**Reply by lucasvsouza28** [reply]
Thank you for the effort! Now i'm receiving another error:

![image](https://github.com/TheEdoRan/next-safe-action/assets/23587471/78829857-e2a7-4443-bc22-1f093cdfc2db)

The action works with validatitons, but build fails

**Reply by TheEdoRan** [reply]
Is this related to [this issue](https://github.com/TheEdoRan/next-safe-action/issues/64#issuecomment-1987478504)?

**Reply by lucasvsouza28** [reply]
With `declaration: false` and `declarationMap: false` in tsconfig the error is gone. Thank you!

**Reply by TheEdoRan** [reply]
Awesome to hear! Just asking out of curiosity, is your Next app in a monorepo? I haven't been able to reproduce this issue with just pnpm, and I still don't know how to fix it for monorepos, unfortunately (as I already said in the issue I linked above, it seems to affect many projects).

**Reply by lucasvsouza28** [reply]
Yes, it is a turbo repo

**Reply by TheEdoRan** [reply]
Ok, thank you for the info. I hope I'll find a solution for that problem eventually. For now, setting those two options in tsconfig.json seems to be the only viable solution.
